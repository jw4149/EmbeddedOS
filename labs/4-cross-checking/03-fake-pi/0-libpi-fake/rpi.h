#ifndef __RPI_H__
#define __RPI_H__
// simplest way to rewrap --- just define replacements in header --- don't
// even need .c
//
// easy, but two downsides: 
//   1. we are polluting the pi program namespace with system includes.
//   2. we aren't using the raw rpi.h so can get different behavior.
#include <stdio.h>
#include <stdlib.h>

// redefine printk as fprintf using variadic macros.
#define printk(args...) fprintf(stderr, args)

static void inline clean_reboot(void) {
    printk("DONE!!!\n");
    exit(0);
}

void notmain(void);
#endif
