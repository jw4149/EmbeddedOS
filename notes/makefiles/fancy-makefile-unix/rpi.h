#include <stdio.h>

#define printk printf
#define debug printf
