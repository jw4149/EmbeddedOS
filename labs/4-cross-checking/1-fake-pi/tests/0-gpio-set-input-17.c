// trivial program to set <pin> as input.  easy to 
// debug.
#include "rpi.h"

enum { pin = 17 };
void notmain(void) {
    output("setting pin %d as an input\n", pin);
    gpio_set_input(pin);
}
