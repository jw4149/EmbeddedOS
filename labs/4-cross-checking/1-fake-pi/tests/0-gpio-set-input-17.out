TRACE:0: calling pi code
TRACE:1: GET32(0x20200004) = 0x643c9869
TRACE:2: PUT32(0x20200004) = 0x641c9869
TRACE: pi exited cleanly: 7 calls to random
